src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -185,6 +189,21 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             }
         }
 
+        internal /*for testing purposes*/ async Task SaveServerExclusionsAsync(
+            IProgressController controller, IProgressStepExecutionEvents notificationEvents,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(controller != null);
+            Debug.Assert(notificationEvents != null);
+
+            notificationEvents.ProgressChanged(Strings.SaveServerExclusionsMessage);
```

Comment:
The message being logged is 
>No file was found under &apos;{0}&apos;.

... but it looks like the message is being logged unconditionally i.e. in the case of the binding being updated, isn't it possible that the file will already exist?